tests/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/HotReload/ProjectHotReloadSessionManagerTests.cs

```diff
@@ -145,6 +145,45 @@ public async Task WhenDebugSymbolsIsFalse_APendingSessionIsNotCreated()
             Assert.False(sessionCreated);
         }
 
+        [Fact]
+        public void WhenNoActiveSession_HasSessionsReturnsFalse()
+        {
+            var capabilities = new[] { "SupportsHotReload" };
+            var propertyNamesAndValues = new Dictionary<string, string?>()
+            {
+                { "TargetFrameworkVersion", "v6.0" },
+                { "DebugSymbols", "true" },
+                // Note: "Optimize" is not included here. The compilers do not optimize by default;
+                // so if the property isn't set that's OK.
+            };
+
+            var activeConfiguredProject = CreateConfiguredProject(capabilities, propertyNamesAndValues);
+            var manager = CreateHotReloadSessionManager(activeConfiguredProject);
+
+            Assert.False(manager.HasActiveHotReloadSessions);
+        }
+
+        [Fact]
+        public async Task WhenActiveSession_HasSessionsReturnsTrue()
+        {
+            var capabilities = new[] { "SupportsHotReload" };
+            var propertyNamesAndValues = new Dictionary<string, string?>()
+            {
+                { "TargetFrameworkVersion", "v6.0" },
+                { "DebugSymbols", "true" },
+                // Note: "Optimize" is not included here. The compilers do not optimize by default;
+                // so if the property isn't set that's OK.
+            };
+
+            var activeConfiguredProject = CreateConfiguredProject(capabilities, propertyNamesAndValues);
+            var manager = CreateHotReloadSessionManager(activeConfiguredProject);
+
+            var environmentVariables = new Dictionary<string, string>();
+            await manager.TryCreatePendingSessionAsync(environmentVariables);
+
+            Assert.False(manager.HasActiveHotReloadSessions);
```

Comment:
Shouldn't this be Assert.True for : WhenActiveSession_HasSessionsReturnsTrue()

Reply:
Yes, I meant to remove that test. That function just creates a pending session but the actual session isn't created until the process starts and ActivateSessionAsync is called. Looking at the  code there is no good way to mock that since there are direct calls to Process.GetProcessId which throws if the process doesn't exist.

Reply:
We only care about active ones since they are the only ones that we can apply changes to